aten/src/ATen/native/native_functions.yaml

```diff
@@ -999,12 +999,19 @@
   variants: function, method
 
 - func: cat(Tensor[] tensors, int dim=0) -> Tensor
+  structured_delegate: cat.out
   dispatch:
-    CompositeExplicitAutograd: cat
+    SparseCPU, SparseCUDA: cat_sparse
```

Comment:
cleaning up the sparse logic :)

Reply:
I guess for consistency we'd like to have an out= variant for sparse kernels, but it looks like that's not true today. It's probably not necessary to try to fix that in this PR.